svid/x509svid/tests/check_svid.c

```diff
@@ -37,13 +54,13 @@ START_TEST(test_x509svid_Parse)
     x509svid_SVID *svid = x509svid_Parse(raw_certs, raw_key, &err);
 
     ck_assert_uint_eq(err, NO_ERROR);
-    ck_assert(svid != NULL);
+    ck_assert_ptr_ne(svid, NULL);
     ck_assert_uint_eq(arrlenu(svid->certs), ITERS);
-    ck_assert(svid->id.td.name != NULL);
-    ck_assert(svid->id.path != NULL);
+    ck_assert_ptr_ne(svid->id.td.name, NULL);
+    ck_assert_ptr_ne(svid->id.path, NULL);
     ck_assert_str_eq(svid->id.td.name, "example.org");
     ck_assert_str_eq(svid->id.path, "/workload-1");
-    ck_assert(svid->private_key != NULL);
+    ck_assert_ptr_ne(svid->private_key, NULL);
 
```

Comment:
needs testing for pemutil_ParseCertificates returning an error inside x509svid_Parse, so we cover freeing certs

Reply:
also, needs testing for passing null values